sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -770,21 +812,112 @@ static int create_sdw_codec_dai(struct device *dev,
 		if (!name)
 			return -ENOMEM;
 
-		cpu_name = devm_kasprintf(dev, GFP_KERNEL,
-					  "SDW%d Pin%d", link_id, j + 2);
-		if (!cpu_name)
-			return -ENOMEM;
+		for (k = 0; k < cpu_num; k++) {
+			int index = SDW_DISCO_LINK_ID(link->adr[k]);
 
-		init_dai_link(dai_links, *id, name, 1 - i, i, cpus, cpu_name,
+			cpu_name = devm_kasprintf(dev, GFP_KERNEL,
+						  "SDW%d Pin%d", index, j + 2);
+			if (!cpu_name)
+				return -ENOMEM;
+
+			cpu_dai[k].dai_name = cpu_name;
 			      codec, link->num_adr, *id, NULL, &sdw_ops);
+
+		init_dai_link(dai_links, *id, name, 1 - i, i, cpu_dai, cpu_num,
 		set_codec_init_func(link, dai_links + *id, 1 - i);
+		cpu_dai += cpu_num;
 		(*id)++;
 		j++;
 	}
 
 	return 0;
 }
 
+static inline bool is_spk(int id)
+{
+	if (id == 0x1308)
+		return true;
+
+	return false;
+}
+
+static int convert_sdw_link_to_sdw_dai(struct device *dev,
+				       struct snd_soc_acpi_mach *mach,
+				       struct sdw_dai_link **sdw_dai,
+				       int *sdw_dai_num)
+{
+	const struct snd_soc_acpi_link_adr *adr_link;
+	const struct snd_soc_acpi_link_adr *link;
+	struct sdw_dai_link *tmp_link;
+	int spk_num, link_num;
+	u64 adr_data[4];
```

Comment:
why 4?

Reply:
it is a tentative setting,  max 4 speaker codec